### PR TITLE
Make update_params=true by default

### LINKED ISSUE #647: Change update method
**Is your feature request related to a problem? Please describe.**
https://github.com/sktime/enhancement-proposals/pull/8

**Describe the solution you'd like**
The `update` method should 
* change to `update_params=True` as default, 
* implement a complete re-fit in the base forecaster as a fall-back option if not overridden in concrete implementations, but raise a warning if fall-back option is used

### PR DESCRIPTION
**Pull request by pabworks**
#### Reference Issues/PRs
Fixes #647. See also sktime/enhancement-proposals#8

#### What does this implement/fix? Explain your changes.
- Makes `update_params=True` by default for all `update`-related forecasting methods.
- Implements a refit in the base Forecaster class when `update_params=True`

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?
- [x] I have set the Theta forecaster `update_params=False` for the `super().update` call since the function then does a custom  routine if `update_params=True`, so, presumably, we don't want to refit. I have also set the `Deseasonalizer` transform to use the updated `y` as opposed to just the passed `y`
- [x] Assumption has been made that `update_predict` on a CV with `step_length>1` when `update_params=True` is not a desired use case for testing since the time-series is not continuous

#### Any other comments
The behaviour of the Forecaster w.r.t. update is as follows:
First, the Forecaster calls `fit` on a time period `y_a=y1,y2,...,yN`
Then, when we call `update`, the Forecaster uses the new information, e.g. `y_b=yN+1,...,yN+k` along with the old information to update the model. If a custom `update` routine is not implemented, then we simply fit a new model on `y=concat(y_a,y_b)`.

Original test failures have been isolated to when `step_length > 1`. In this case, when `update_params=True`, the model attempts to refit on a concatenation of the previous data (e.g. daily from "2000-01-01" to "2000-02-01") with the new data point as selected by the CV splitter. If the CV splitter has a `step_length > 1` then the frequency of the time series is no longer valid (e.g. daily up to "2000-02-01" then stepping multiple days into the future, e.g. "2000-02-07", before continuing at the same frequency)

The testing of `update_predict` where `step_length>1` and `update_params=True` has been left out, as a result.

#### PR checklist

- [x] set `update_params=True` for all methods
- [x] ensure all tests still pass when `update_params=False` (maintain original behaviour)
- [x] implement refit method in base Forecaster using original and update data
- [x] ensure all tests pass when `update_params=True` and `step_length=1`
- [ ] check desired behaviour for `update_params=True` and `step_length>1`

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.

### CODE REVIEW

File: sktime/forecasting/compose/_reduce.py [hunk 1]
```diff
@@ -43,23 +45,32 @@ def __init__(self, regressor, window_length=10, step_length=1):
         self.step_length_ = None
         self._cv = None
 
-    def update(self, y, X=None, update_params=False):
+    def update(self, y, X=None, update_params=True):
```
Comment (mloning) [inline sktime/forecasting/compose/_reduce.py:48]: Thinking about this, can we not simply remove this here as we inherit `update` from the base forecaster? I probably overrode it to handle the error when X is passed, but this should already be handled in `fit` and `update` only works on fitted forecasters, so we should be able to simply remove it here, no?

**Review by mloning — COMMENTED** [review]
Thanks @pabworks - looks all good! I left one comment which we can perhaps resolve in this PR too if it works. Otherwise happy to merge it!

**Comment by mloning** [comment]
@pabworks I'll work on this today to make a few more minor changes.

**Review by mloning — APPROVED** [review]

**Comment by pabworks** [comment]
Thanks @mloning! (I was off ill last week so didn't see your comments until this morning, sorry!)

**Comment by mloning** [comment]
@pabworks no worries - thanks for doing most of the work on this one!

**Pull request merged**